lib/perl/Genome/Site/TGI/Extension/Sys.pm

```diff
@@ -69,6 +71,82 @@ sub get_lsf_job_status {
     return $status;
 }
 
+sub bsub_and_wait_for_completion {
+    my($class, %bsub_args) = @_;
+
+    my $special_args;
+    ($special_args, %bsub_args) = _bsub_and_wait_for_completion__validate_args(%bsub_args);
+    my($commands, $on_submit_cb, $on_complete_cb) = @$special_args{'commands','on_submit','on_complete'};
+
```

Comment:
I did it this way to make calling it simpler by having the regular bsub args (like ```queue``` and ```job_group```) passed together with the args for this method (like ```cmds``` and ```on_submit```).  Otherwise, you'd end up with a more complicated function call like this
```
Genome::Sys->bsub_and_wait_completion(
    { queue => 'short',
      job_group => 'blah',
    },
    {  cmds => [ '/bin/true', 'bin/false'],
      on_submit => sub { ... },
    }
);